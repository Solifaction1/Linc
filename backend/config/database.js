const path = require('path');

const connection = process.env.NODE_ENV !== 'production' ?
  {
    client: 'sqlite',
    connection: {
      filename: path.join(__dirname, '..', '.tmp/data.db'),
    },
    useNullAsDefault: true,
  } : {
    client: "postgres",
    acquireConnectionTimeout: 120000,
    connection: {
      // host: `/cloudsql/${process.env.INSTANCE_CONNECTION_NAME}`, // for a PostgreSQL database
      host: process.env.DATABASE_HOST, // for a PostgreSQL database
      database: process.env.DATABASE_NAME,
      user: process.env.DATABASE_USERNAME,
      password: process.env.DATABASE_PASSWORD,
      ssl: false,

    },
    useNullAsDefault: true,
    debug: false,
  };

module.exports = ({ }) => ({
  connection: connection,
});
