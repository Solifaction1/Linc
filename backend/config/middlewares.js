module.exports = [
  'strapi::errors',
  {
    name: 'strapi::cors',
    config: {
      enabled: true,
      headers: ['*'],
      origin: ['*']
    }
  },
  'strapi::poweredBy',
  'strapi::logger',
  'strapi::query',
  'strapi::body',
  'strapi::session',
  'strapi::favicon',
  'strapi::public',
  {
    name: 'strapi::security',
    config: {
      contentSecurityPolicy: {
        useDefaults: true,
        directives: {
          'connect-src': ["'self'", 'https:'],
          'img-src': [
            "'self'",
            'data:',
            'blob:',
            'dl.airtable.com',
            "cdn.jsdelivr.net",
            "strapi.io",
            "s3.amazonaws.com",
            "editor.unlayer.com",
            process.env.CDN_URL,
          ],
          "script-src": ["'self'", "editor.unlayer.com"],
          "frame-src": ["'self'", "editor.unlayer.com"],
          'media-src': [
            "'self'",
            'data:',
            'blob:',
            'dl.airtable.com',
            process.env.CDN_URL,
          ],
          upgradeInsecureRequests: null,
        },
      },
    },
  },
];
