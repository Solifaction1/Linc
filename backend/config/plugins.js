module.exports = ({ env }) => ({
  "strapi-plugin-populate-deep": {
    config: {
      defaultDepth: 5, // Default is 5
    },
  },
  "duplicate-button": true,
  email: {
    config: {
      provider: "mailgun",
      providerOptions: {
        key: env("MAILGUN_API_KEY"), // Required
        domain: env("MAILGUN_DOMAIN"), // Required
        url: "https://api.eu.mailgun.net", //Optional. If domain region is Europe use 'https://api.eu.mailgun.net'
      },
      settings: {
        defaultFrom: "<mg@linc.legal>",
        defaultReplyTo: "<mg@linc.legal>",
      },
    },
  },
  upload: {
    config: {
      provider: "aws-s3",
      providerOptions: {
        accessKeyId: env("AWS_ACCESS_KEY_ID"),
        secretAccessKey: env("AWS_ACCESS_SECRET"),
        region: env("AWS_REGION"),
        params: {
          Bucket: env("AWS_BUCKET"),
        },
      },
      actionOptions: {
        upload: {},
        uploadStream: {},
        delete: {},
      },
    },
  },
  "content-versioning": {
    enabled: false,
  },
  slugify: {
    enabled: false,
    config: {
      contentTypes: {
        blog: {
          field: "slug",
          references: "title",
        },
      },
    },
  },
  "preview-button": {
    enabled: true,
    resolve: "./src/plugins/preview-button", // path to plugin folder
  },
  "email-designer": {
    enabled: true,

    // ⬇︎ Add the config property
    config: {
      editor: {
        // optional - if you have a premium unlayer account
        // projectId: [UNLAYER_PROJECT_ID],

        tools: {
          heading: {
            properties: {
              text: {
                value: "This is the new default text!",
              },
            },
          },
        },
        options: {
          features: {
            colorPicker: {
              presets: ["#D9E3F0", "#F47373", "#697689", "#37D67A"],
            },
          },
          fonts: {
            showDefaultFonts: false,
            /*
             * If you want use a custom font you need a premium unlayer account and pass a projectId number :-(
             */
            customFonts: [
              {
                label: "Anton",
                value: "'Anton', sans-serif",
                url: "https://fonts.googleapis.com/css?family=Anton",
              },
              {
                label: "Lato",
                value: "'Lato', Tahoma, Verdana, sans-serif",
                url: "https://fonts.googleapis.com/css?family=Lato",
              },
              // ...
            ],
          },
          mergeTags: [
            {
              name: "Naam",
              value: "{{= USER.Naam }}",
              sample: "Naam",
            },
            {
              name: "Voornaam",
              value: "{{= USER.Voornaam }}",
              sample: "Voornaam",
            },
            {
              name: "Email",
              value: "{{= USER.Email }}",
              sample: "john@doe.com",
            },
            {
              name: "Telefoonnummer",
              value: "{{= USER.Telefoonnummer }}",
              sample: "0123456789",
            },
            {
              name: "select",
              value: "{{= USER.select }}",
              sample: "select",
            },
            {
              name: "textArea",
              value: "{{= USER.textArea }}",
              sample: "textArea",
            },
            {
              name: "Form ID",
              value: "{{= USER.formId }}",
              sample: "0",
            },
          ],
        },
        appearance: {
          theme: "dark",
          panels: {
            tools: {
              dock: "left",
            },
          },
        },
      },
    },
  },
  seo: {
    enabled: true,
  },
  ckeditor: {
    enabled: true,
    config: {
      plugin: {
        // disable data-theme tag setting //
        // setAttribute:false,
        // disable strapi theme, will use default ckeditor theme //
        // strapiTheme:false,
        // styles applied to editor container (global scope) //
        // styles:`
        // :root{
        //   --ck-color-focus-border:red;
        //   --ck-color-text:red;
        // }
        // `
        styles: `
        .ck-editor__main p,
        .ck.ck-list__item .ck-button .ck-button__label {
          line-height: normal;
        }`,
      },
      editor: {
        // editor default config

        // https://ckeditor.com/docs/ckeditor5/latest/features/markdown.html
        // if you need markdown support and output set: removePlugins: [''],
        // default is
        removePlugins: ["Markdown"],

        // https://ckeditor.com/docs/ckeditor5/latest/features/toolbar/toolbar.html
        toolbar: {
          items: [
            "paragraph",
            "heading1",
            "heading2",
            "heading3",
            "|",
            "bold",
            "italic",
            "underline",
            "fontSize",
            "removeFormat",
            "|",
            "bulletedList",
            "numberedList",
            "alignment",
            "|",
            "StrapiMediaLib",
            "link",
            "horizontalLine",
            "|",
            "sourceEditing",
            "fullScreen",
            "undo",
            "redo",
          ],
        },
        // https://ckeditor.com/docs/ckeditor5/latest/features/font.html
        fontSize: {
          options: [
            9,
            11,
            13,
            {
              title: "default",
              model: "18px",
            },
            17,
            19,
            21,
            27,
            35,
          ],
          supportAllValues: false,
        },
        fontFamily: {
          options: [
            "default",
            "Arial, Helvetica Neue, Helvetica, Source Sans Pro, sans-serif",
            "Courier New, Courier, monospace",
            "Georgia, serif",
            "Lucida Sans Unicode, Lucida Grande, sans-serif",
            "Tahoma, Geneva, sans-serif",
            "Times New Roman, Times, serif",
            "Trebuchet MS, Helvetica, sans-serif",
            "Verdana, Geneva, sans-serif",
            "Roboto, Roboto Black, Roboto Medium, Roboto Light, sans-serif",
          ],
          supportAllValues: true,
        },
        fontColor: {
          columns: 5,
          documentColors: 10,
        },
        fontBackgroundColor: {
          columns: 5,
          documentColors: 10,
        },
        // https://ckeditor.com/docs/ckeditor5/latest/features/ui-language.html
        // default language: 'en',
        // https://ckeditor.com/docs/ckeditor5/latest/features/images/images-overview.html
        image: {
          resizeUnit: "%",
          resizeOptions: [
            {
              name: "resizeImage:original",
              value: null,
              icon: "original",
            },
            {
              name: "resizeImage:25",
              value: "25",
              icon: "small",
            },
            {
              name: "resizeImage:50",
              value: "50",
              icon: "medium",
            },
            {
              name: "resizeImage:75",
              value: "75",
              icon: "large",
            },
          ],
          toolbar: [
            "toggleImageCaption",
            "imageTextAlternative",
            "imageStyle:inline",
            "imageStyle:block",
            "imageStyle:side",
            "linkImage",
            "resizeImage:25",
            "resizeImage:50",
            "resizeImage:75",
            "resizeImage:original",
          ],
        },
        // https://ckeditor.com/docs/ckeditor5/latest/features/table.html
        table: {
          contentToolbar: [
            "tableColumn",
            "tableRow",
            "mergeTableCells",
            "tableCellProperties",
            "tableProperties",
            "toggleTableCaption",
          ],
        },
        // https://ckeditor.com/docs/ckeditor5/latest/features/headings.html
        heading: {
          options: [
            {
              model: "paragraph",
              title: "Paragraph",
              class: "ck-heading_paragraph",
            },
            {
              model: "heading1",
              view: "h1",
              title: "Heading 1",
              class: "ck-heading_heading1",
            },
            {
              model: "heading2",
              view: "h2",
              title: "Heading 2",
              class: "ck-heading_heading2",
            },
            {
              model: "heading3",
              view: "h3",
              title: "Heading 3",
              class: "ck-heading_heading3",
            },
          ],
        },
        // https://ckeditor.com/docs/ckeditor5/latest/features/general-html-support.html
        htmlSupport: {
          allow: [
            {
              name: "img",
              attributes: {
                sizes: true,
                loading: true,
              },
            },
          ],
        },
      },
    },
  },
});
