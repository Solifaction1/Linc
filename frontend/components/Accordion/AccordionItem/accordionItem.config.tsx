import AccordionData from '@models/accordion.model'

const defaultProps: AccordionData = {
	title: 'title',
}

export { defaultProps }
