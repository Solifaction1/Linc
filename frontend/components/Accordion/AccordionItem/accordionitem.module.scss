@import "styles/vars";
@import "styles/utilities";

.section {
    overflow: hidden;

    width: 70vw;
    margin: 0 auto;

    transition: height $animation-in ease;

    &.active{
        overflow: unset;
    }

    .accordionButton {
        position: relative;
        display: flex;
        align-items: center;
        justify-content: space-between;

        width: 100%;
        min-height: calc(#{$spacing-base} * 11);
        @media (max-width: $break-md) {
            min-height: calc(#{$spacing-base} * 8);
        }

        margin-bottom: calc(#{$spacing-base} * 4);
        border: none;
        border-top: 1px solid $color-beige-dark;

        background-color: transparent;

        cursor: pointer;

        .title {
            color: $color-beige-dark;

            font-size: $font-size-title-md-m;
            font-weight: $font-weight-regular;
            text-align: start;
            line-height: 3.5rem;

            @media (max-width: $break-md) {
                font-size: $font-size-title-xs;
                line-height: 2.25rem;
            }
        }

        .arrowWrapper {
            position: relative;
            min-width: calc(#{$spacing-base} * 6);
            min-height: calc(#{$spacing-base} * 6);


            svg {
                width: 5rem;
                height: 5rem;
                fill: $color-green-dark;
            }
        }

        .arrow {
            position: absolute;
            top: 50%;
            left: 50%;

            fill: $color-beige-dark;

            transform: translate(-50%, -50%);
        }
    }

    .accordionContainer {
        position: relative;
        z-index: 1;

        padding-bottom: calc(#{$spacing-base} * 6);
        border-top: unset;
		
        line-height: calc($font-size-text-sm * 1.4);
		
        p {
            font-size: $font-size-text-xlg;

            @media (max-width: $break-md) {
                font-size: $font-size-text-sm;
            }
        }

        img {
            @media (min-width: $break-md) {
                max-width: 75%;
            }
        }
    }
}
