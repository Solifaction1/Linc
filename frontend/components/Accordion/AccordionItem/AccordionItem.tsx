import React, { useRef, useEffect, useState } from 'react'

import styles from '@components/Accordion/AccordionItem/accordionitem.module.scss'
import SwitchIcon from '@templates/SwitchIcon'
import AccordionData from '@models/accordion.model'
import { Icon } from '@templates/switch-icon.config'

const AccordionItem: React.FC<AccordionData> = (props: AccordionData) => {
	const buttonRef = useRef<HTMLButtonElement>(null)
	const bodyRef = useRef<HTMLDivElement>(null)
	const [isActive, setIsActive] = useState<boolean | null>(null)

	const [height, setHeight] = useState(0)

	const validateHeight = () => {
		const buttonHeight = buttonRef.current?.clientHeight ?? 0
		const bodyHeight = bodyRef.current?.clientHeight ?? 0
		setHeight(isActive ? buttonHeight + bodyHeight : buttonHeight)
	}

	useEffect(() => {
		validateHeight()
		window.addEventListener('resize', () => validateHeight())
	}, [isActive, height])

	return (
		<div>
			<div
				className={`${styles.section} ${isActive ? styles.active : ''}`}
				key={props.title}
				style={{ height: `${height}px` }}
			>
				<button
					type="button"
					ref={buttonRef}
					className={styles.accordionButton}
					onClick={() => setIsActive(!isActive)}
				>
					<h4 className={styles.title}>{props.title}</h4>

					<div className={styles.arrowWrapper}>
						<SwitchIcon
							className={`${isActive && styles.active} ${styles.arrow}`}
							icon={isActive ? Icon.ANGLE_UP : Icon.ANGLE_DOWN}
						/>
					</div>
				</button>
			</div>
			<div
				style={{ display: isActive ? 'unset' : 'none' }}
				ref={bodyRef}
				className={styles.accordionContainer}
			>
				{props.children}
			</div>
		</div>
	)
}

export default AccordionItem
