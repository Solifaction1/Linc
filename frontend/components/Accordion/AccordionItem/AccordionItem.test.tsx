import React from 'react'

import { render } from '@testing-library/react'
import '@testing-library/jest-dom'
import AccordionItem from './AccordionItem'
import { defaultProps } from './accordionItem.config'

describe('AccordionItem', () => {
	it('renders an AccordionItem', () => {
		const { container, getByText } = render(<AccordionItem {...defaultProps} />)

		expect(container).toMatchSnapshot()

		expect(container.getElementsByClassName('section')[0]).toBeVisible()

		const accordionButton = container.querySelector('button')
		const accordionArrowWrapper =
			container.getElementsByClassName('arrowWrapper')[0]
		const accordionContainer =
			container.getElementsByClassName('accordionContainer')[0]

		expect(accordionButton?.getAttribute('class')).toContain('accordionButton')
		expect(accordionButton?.getAttribute('type')).toContain('button')

		expect(
			accordionButton?.contains(container.getElementsByClassName('title')[0])
		)
		expect(getByText('title', { selector: 'h4' })).toBeVisible()

		expect(accordionButton?.contains(accordionArrowWrapper))
		expect(accordionArrowWrapper?.contains(container.querySelector('svg')))
		expect(accordionArrowWrapper?.contains(container.querySelector('path')))

		expect(
			accordionContainer?.contains(
				container.getElementsByClassName('ygwys undefined undefined')[0]
			)
		)
	})
})
