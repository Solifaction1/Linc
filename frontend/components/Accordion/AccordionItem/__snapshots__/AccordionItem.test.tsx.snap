// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccordionItem renders an AccordionItem 1`] = `
<div>
  <div>
    <div
      class="section "
      style="height: 0px;"
    >
      <button
        class="accordionButton"
        type="button"
      >
        <h4
          class="title"
        >
          title
        </h4>
        <div
          class="arrowWrapper"
        >
          <svg
            class="null arrow"
            fill="currentColor"
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M17,9.17a1,1,0,0,0-1.41,0L12,12.71,8.46,9.17a1,1,0,0,0-1.41,0,1,1,0,0,0,0,1.42l4.24,4.24a1,1,0,0,0,1.42,0L17,10.59A1,1,0,0,0,17,9.17Z"
            />
          </svg>
        </div>
      </button>
    </div>
    <div
      class="accordionContainer"
      style="display: none;"
    />
  </div>
</div>
`;
